### ISSUE TITLE
Error when 'flutter run': Undefined symbols for architecture arm64:

### ISSUE THREAD

**Issue by Charles1198** [flutter/flutter#41900]
I get error when 'flutter run':
```
charlesdeMBP:rongchang_app charles$ flutter run
Launching lib/main.dart on yueji的 iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: G5JFP3Y63E
Running pod install...                                              2.8s
Running Xcode build...                                                  
                                                   
Xcode build done.                                           10.2s
Failed to build iOS app
Error output from Xcode build:
↳
    2019-10-03 18:01:56.053 xcodebuild[60982:3066650]  DTDeviceKit: deviceType from 43905bacecbd9af724d1c568192bbbe425c383b4 was NULL
    2019-10-03 18:01:56.115 xcodebuild[60982:3066698]  DTDeviceKit: deviceType from 43905bacecbd9af724d1c568192bbbe425c383b4 was NULL
    ** BUILD FAILED **


Xcode's output:
↳
    Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_FlutterStandardMessageCodec", referenced from:
          objc-class-ref in FlutterWebView.o
      "_OBJC_CLASS_$_FlutterError", referenced from:
          objc-class-ref in FLTWKNavigationDelegate.o
          objc-class-ref in FlutterWebView.o
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in FLTCookieManager.o
          objc-class-ref in FlutterWebView.o
      "_FlutterMethodNotImplemented", referenced from:
          -[FLTCookieManager handleMethodCall:result:] in FLTCookieManager.o
          ___83-[FLTWKNavigationDelegate webView:decidePolicyForNavigationAction:decisionHandler:]_block_invoke in FLTWKNavigationDelegate.o
          -[FLTWebViewController onMethodCall:result:] in FlutterWebView.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build systemnote: Planning buildnote: Constructing build description

Could not build the precompiled application for the device.

Error launching application on yueji的 iPhone.
```

It seems not `FlutterWebView` cause the problem, because I have many dependencies in my application, when I remove `FlutterWebView` and `flutter run` get the same error but with other library.

`flutter doctor -v` show this:
```
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.6 18G95, locale zh-Hans-CN)
    • Flutter version 1.9.1+hotfix.4 at /Users/charles/development/flutter
    • Framework revision cc949a8e8b (6 days ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/charles/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.8.0

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.38.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (1 available)
    • yueji的 iPhone • 43905bacecbd9af724d1c568192bbbe425c383b4 • ios • iOS 12.4.1

• No issues found!
```

Thank you very much for any help!

**Comment by BondarenkoStas** [comment]
@Charles1198 Please check if you can run it on simulator. Can you please try if it's the same on master channel?
Seems similar to #38746

**Comment by Charles1198** [comment]
> @Charles1198 Please check if you can run it on simulator. Can you please try if it's the same on master channel?
> Seems similar to #38746

On simulator the same error but " Undefined symbols for architecture x86_64:"

**Comment by Charles1198** [comment]
> @Charles1198 Please check if you can run it on simulator. Can you please try if it's the same on master channel?
> Seems similar to #38746

There has no error 3 days age, I have published both android and iOS app successfully and not open the project until this morning. Today when I try to run the application again, it runs error. It is so weird.

**Comment by Charles1198** [comment]
Solved by check flutter channel from stable to master.

**Comment by timtraversy** [comment]
I'm hitting this too and I think it should be reopened. Build shouldn't be failing on stable.

**Comment by BondarenkoStas** [comment]
@timtraversy Can you please check if you having the issue on master? (to make sure it's the same issue). The fixes landed on master eventually will be in stable, that's the process of development. Hotfixes to stable are done only for some very severe cases, and in general not a good practice.
I understand your pain, and if you can't use dev and stable also blocks you, I can help to find the fix commits, so you can cherry-pick them into your stable.

**Comment by pedrobrasileiro** [comment]
> Solved by check flutter channel from stable to master.

Dont work from me...

My mac as High Sierra (10.13.6) and continue this error.

This project i created with flutter create name_project

```
$ flutter doctor -v
[✓] Flutter (Channel stable, v1.9.1+hotfix.6, on Mac OS X 10.13.6 17G8037, locale en-BR)
    • Flutter version 1.9.1+hotfix.6 at /Users/unimedmobile/Desenvolvimento/libs/flutter
    • Framework revision 68587a0916 (6 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/unimedmobile/Desenvolvimento/libs/android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/unimedmobile/Desenvolvimento/libs/android-sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • CocoaPods version 1.8.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 40.0.2
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.39.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • SM G9600 • 27c2c7b199217ece • android-arm64 • Android 9 (API 28)

• No issues found!
```



```
Encountered error while building for device.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      BuildIOSCommand.runCommand (package:flutter_tools/src/commands/build_ios.dart:94:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#3      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#4      _rootRunUnary (dart:async/zone.dart:1132:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#9      Future._completeWithValue (dart:async/future_impl.dart:522:5)
#10     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

**Comment by BondarenkoStas** [comment]
@pedrobrasileiro You posted doctor with stable, if it doesn't work with master channel then probably there is another cause. Can you please file a new issue with that, including full error log (probably referencing this issue)? In this way things will be less messy and author won't be notified for nothing.

**Comment by nicowernli** [comment]
Still happens to me when I try to use flutter flavors:

```bash
 $ flutter doctor -v
[✓] Flutter (Channel master, v1.12.17-pre.80, on Mac OS X 10.15.1 19B88, locale es-ES)
    • Flutter version 1.12.17-pre.80 at /Users/nicowernli/Development/lib/flutter
    • Framework revision 649e188562 (7 hours ago), 2019-12-05 09:41:49 +0100
    • Engine revision fdaa7cf121
    • Dart version 2.7.0 (build 2.7.0-dev.2.1 413867d678)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/nicowernli/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/nicowernli/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.2.1, Build version 11B500
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 41.1.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • iPhone 11 Pro Max • 86AEAB75-B219-470A-B54F-547316D93AFC • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-2 (simulator)

• No issues found!
 ```

The error: 
```
d: warning: ignoring file /Users/nicowernli/Development/lib/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter, missing required architecture i386 in file /Users/nicowernli/Development/lib/flutter/bin/cache/artifacts/engine/ios/Flutter.framework/Flutter (3 slices)
Undefined symbols for architecture i386:
  "_FlutterMethodNotImplemented", referenced from:
      ___52+[FLTSharedPreferencesPlugin registerWithRegistrar:]_block_invoke in SharedPreferencesPlugin.o
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in SharedPreferencesPlugin.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Charles1198** [comment]
I met this problem several times later, every time I can solve it by checkout to master channel and then checkout to stable channel. I know it is not the right solution but it works. Hope to help you.

**Comment by mayurdhurpate** [comment]
Faced this error recently after upgrading from Flutter 1.9 to Flutter 1.12. Not sure if this is because of the Smartlook plugin or a general Flutter issue. Also, many times the Flutter run command succeeds on iOS emulator and fails other times. Also, I deleted the previous Podfile and let Flutter generate the new Podfile and new Podfile.lock.

Output of `flutter run -v`: https://pastebin.com/t0G8rEnB

Error snippet:
```
/Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
                            uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook.LinkFileList -install_name
                            @rpath/smartlook.framework/smartlook -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath
-Xlinker
                            @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker

/Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
                            uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook_lto.o -Xlinker -export_dynamic
-Xlinker
                            -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -framework Flutter
-framework Smartlook
                            -framework Foundation -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker

/Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
                            uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook_dependency_info.dat -o

/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/smartlook/smartlook.framework/smartlook
                        Undefined symbols for architecture x86_64:
                          "_OBJC_CLASS_$_Smartlook", referenced from:
                              objc-class-ref in SmartlookPlugin.o
                             (maybe you meant: _OBJC_CLASS_$_SmartlookPlugin)
                          "_SLNavigationTypeExit", referenced from:
                              -[SmartlookPlugin _internal_handleMethodCall:result:] in SmartlookPlugin.o
                          "_SLNavigationTypeEnter", referenced from:
                              -[SmartlookPlugin _internal_handleMethodCall:result:] in SmartlookPlugin.o
                        ld: symbol(s) not found for architecture x86_64
                        clang: error: linker command failed with exit code 1 (use -v to see invocation)
                    
                        Ld

/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter/webview_flutter.framework/webview_flu
                        tter normal x86_64 (in target 'webview_flutter' from project 'Pods')
                            cd /Users/mayurdhurpate/code/pruoo_app/pruoo_app/ios/Pods
                            /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target
                            x86_64-apple-ios8.0-simulator -dynamiclib -isysroot

/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk
                            -L/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter
                            -F/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter
                            -F/Users/mayurdhurpate/code/pruoo_app/pruoo_app/ios/Pods/../Flutter -filelist
```


Flutter Doctor

```
Mayurs-MacBook-Air:firebase_admin mayurdhurpate$ flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale en-IN)
    • Flutter version 1.12.13+hotfix.5 at /Users/mayurdhurpate/code/source/flutter
    • Framework revision 27321ebbad (4 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/mayurdhurpate/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • iPhone 11 • 2FA502F5-1E21-4074-B410-FF5A8869F188 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
```

**Comment by kushalmahapatro** [comment]
Facing the same issue, till yesterday it was running fine, but suddenly today morning, it gives out the below error,
`Launching lib/main.dart on Satya P’s iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: MZ79BCV5W9
Running Xcode build...
Xcode build done.                                            7.3s
Failed to build iOS app
Error output from Xcode build:
↳
    2020-01-19 16:02:38.229 xcodebuild[28815:213254]  DTDeviceKit: deviceType from c5b1d78e3385e328d467b7568cab8c71c167e66b was NULL
    2020-01-19 16:02:38.289 xcodebuild[28815:213254]  DTDeviceKit: deviceType from c5b1d78e3385e328d467b7568cab8c71c167e66b was NULL
    ** BUILD FAILED **


Xcode's output:
↳
    Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_FlutterStandardMessageCodec", referenced from:
          objc-class-ref in FlutterWebView.o
      "_OBJC_CLASS_$_FlutterError", referenced from:
          objc-class-ref in FLTWKNavigationDelegate.o
          objc-class-ref in FlutterWebView.o
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in FLTCookieManager.o
          objc-class-ref in FlutterWebView.o
      "_FlutterMethodNotImplemented", referenced from:
          -[FLTCookieManager handleMethodCall:result:] in FLTCookieManager.o
          ___83-[FLTWKNavigationDelegate webView:decidePolicyForNavigationAction:decisionHandler:]_block_invoke in FLTWKNavigationDelegate.o
          -[FLTWebViewController onMethodCall:result:] in FlutterWebView.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Planning build
    note: Constructing build description

Could not build the precompiled application for the device.

Error launching application on Satya P’s iPhone.`

Flutter Doctor:
`kushal@Kushals-MacBook-Pro StyliShop_Mobile % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale en)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.41.1)
[✓] Connected device (2 available)

• No issues found!
`

Tried switching to master still, the issue persists.

**Comment by dannyalbuquerque** [comment]
> Faced this error recently after upgrading from Flutter 1.9 to Flutter 1.12. Not sure if this is because of the Smartlook plugin or a general Flutter issue. Also, many times the Flutter run command succeeds on iOS emulator and fails other times. Also, I deleted the previous Podfile and let Flutter generate the new Podfile and new Podfile.lock.
> 
> Output of `flutter run -v`: https://pastebin.com/t0G8rEnB
> 
> Error snippet:
> 
> ```
> /Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
>                             uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook.LinkFileList -install_name
>                             @rpath/smartlook.framework/smartlook -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath
> -Xlinker
>                             @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker
> 
> /Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
>                             uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook_lto.o -Xlinker -export_dynamic
> -Xlinker
>                             -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -framework Flutter
> -framework Smartlook
>                             -framework Foundation -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker
> 
> /Users/mayurdhurpate/Library/Developer/Xcode/DerivedData/Runner-dmcarvomewvuhqgxwbxwwabloblz/Build/Intermediates.noindex/Pods.b
>                             uild/Debug-iphonesimulator/smartlook.build/Objects-normal/x86_64/smartlook_dependency_info.dat -o
> 
> /Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/smartlook/smartlook.framework/smartlook
>                         Undefined symbols for architecture x86_64:
>                           "_OBJC_CLASS_$_Smartlook", referenced from:
>                               objc-class-ref in SmartlookPlugin.o
>                              (maybe you meant: _OBJC_CLASS_$_SmartlookPlugin)
>                           "_SLNavigationTypeExit", referenced from:
>                               -[SmartlookPlugin _internal_handleMethodCall:result:] in SmartlookPlugin.o
>                           "_SLNavigationTypeEnter", referenced from:
>                               -[SmartlookPlugin _internal_handleMethodCall:result:] in SmartlookPlugin.o
>                         ld: symbol(s) not found for architecture x86_64
>                         clang: error: linker command failed with exit code 1 (use -v to see invocation)
>                     
>                         Ld
> 
> /Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter/webview_flutter.framework/webview_flu
>                         tter normal x86_64 (in target 'webview_flutter' from project 'Pods')
>                             cd /Users/mayurdhurpate/code/pruoo_app/pruoo_app/ios/Pods
>                             /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target
>                             x86_64-apple-ios8.0-simulator -dynamiclib -isysroot
> 
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk
>                             -L/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter
>                             -F/Users/mayurdhurpate/code/pruoo_app/pruoo_app/build/ios/Debug-iphonesimulator/webview_flutter
>                             -F/Users/mayurdhurpate/code/pruoo_app/pruoo_app/ios/Pods/../Flutter -filelist
> ```
> 
> Flutter Doctor
> 
> ```
> Mayurs-MacBook-Air:firebase_admin mayurdhurpate$ flutter doctor -v
> [✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale en-IN)
>     • Flutter version 1.12.13+hotfix.5 at /Users/mayurdhurpate/code/source/flutter
>     • Framework revision 27321ebbad (4 weeks ago), 2019-12-10 18:15:01 -0800
>     • Engine revision 2994f7e1e6
>     • Dart version 2.7.0
> 
>  
> [✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
>     • Android SDK at /Users/mayurdhurpate/Library/Android/sdk
>     • Android NDK location not configured (optional; useful for native profiling support)
>     • Platform android-28, build-tools 28.0.3
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 11.3)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 11.3, Build version 11C29
>     • CocoaPods version 1.8.4
> 
> [✓] Android Studio (version 3.3)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin version 33.3.1
>     • Dart plugin version 182.5215
>     • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
> 
> [✓] VS Code (version 1.41.1)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.7.1
> 
> [✓] Connected device (1 available)
>     • iPhone 11 • 2FA502F5-1E21-4074-B410-FF5A8869F188 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)
> ```

@mayurdhurpate I'm facing the same issue with Smartlook, did you find a solution ?

**Comment by baoxu** [comment]
删除Podfile，Podfile.lock，Pods文件夹，Runner.xcworkspace，让flutter重新生成这几个文件，就好了

**Comment by waitwalker** [comment]
> 删除Podfile，Podfile.lock，Pods文件夹，Runner.xcworkspace，让flutter重新生成这几个文件，就好了

Right. Absolutely right.

**Comment by emiliodallatorre** [comment]
> 删除Podfile，Podfile.lock，Pods文件夹，Runner.xcworkspace，让flutter重新生成这几个文件，就好了

I tried also by only running `flutter create .`: it worked like a charm!
The problem was in some of the XCode project files.

**Comment by kushalmahapatro** [comment]
> Facing the same issue, till yesterday it was running fine, but suddenly today morning, it gives out the below error,
> 
> ```
> Launching lib/main.dart on Satya P’s iPhone in debug mode...
> Automatically signing iOS for device deployment using specified development team in Xcode project: MZ79BCV5W9
> Running Xcode build...
> Xcode build done.                                            7.3s
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     2020-01-19 16:02:38.229 xcodebuild[28815:213254]  DTDeviceKit: deviceType from c5b1d78e3385e328d467b7568cab8c71c167e66b was NULL
>     2020-01-19 16:02:38.289 xcodebuild[28815:213254]  DTDeviceKit: deviceType from c5b1d78e3385e328d467b7568cab8c71c167e66b was NULL
>     ** BUILD FAILED **
> 
> 
> Xcode's output:
> ↳
>     Undefined symbols for architecture arm64:
>       "_OBJC_CLASS_$_FlutterStandardMessageCodec", referenced from:
>           objc-class-ref in FlutterWebView.o
>       "_OBJC_CLASS_$_FlutterError", referenced from:
>           objc-class-ref in FLTWKNavigationDelegate.o
>           objc-class-ref in FlutterWebView.o
>       "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
>           objc-class-ref in FLTCookieManager.o
>           objc-class-ref in FlutterWebView.o
>       "_FlutterMethodNotImplemented", referenced from:
>           -[FLTCookieManager handleMethodCall:result:] in FLTCookieManager.o
>           ___83-[FLTWKNavigationDelegate webView:decidePolicyForNavigationAction:decisionHandler:]_block_invoke in FLTWKNavigationDelegate.o
>           -[FLTWebViewController onMethodCall:result:] in FlutterWebView.o
>     ld: symbol(s) not found for architecture arm64
>     clang: error: linker command failed with exit code 1 (use -v to see invocation)
>     note: Using new build system
>     note: Planning build
>     note: Constructing build description
> 
> Could not build the precompiled application for the device.
> 
> Error launching application on Satya P’s iPhone.
> ```
> 
> Flutter Doctor:
> 
> ```
> kushal@Kushals-MacBook-Pro StyliShop_Mobile % flutter doctor
> Doctor summary (to see all details, run flutter doctor -v):
> [✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale en)
>  
> [✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
> [✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
> [✓] Android Studio (version 3.5)
> [✓] VS Code (version 1.41.1)
> [✓] Connected device (2 available)
> 
> • No issues found!
> ```
> 
> Tried switching to master still, the issue persists.

This issue was fixed, 
Actually my bad, I opened the Runner.xcodeproj instead of Runner.xcworkspace. 
Opening the Runner.xcworkspace and building from it didn't gave me any issue, yeah but running from flutter used to give me. 

So I opened the Runner.xcworkspace and ran from XCode and after the successful build, the issue disappeared from both XCode as well as Flutter.

**Comment by scullabyte** [comment]
This is a VERY annoying issue. I have been developing everything without a problem for four months. No errors on builds. Nothing. Ever since upgrading flutter, everything has been halted for about 5 days now. I've used every channel. Tried every fix on every possible thread. NOTHING. I'm about to completely give up on flutter and the 4 months I've spent developing this app. THE PLATFORM IS TOO UNSTABLE.

**Comment by vargavince91** [comment]
This comment as a command (please make sure you understand what you are deleting):

```
flutter clean && rm ios/Podfile ios/Podfile.lock pubspec.lock && rm -rf ios/Pods ios/Runner.xcworkspace && flutter run
```

**Comment by bdolewski** [comment]
@vargavince91 had the same issue as @Charles1198. You combo-cleanup fixed my issue. However when I rebuild the app I've got unexpectedly up-to-date pods that wouldn't upgrade before. At least it's working :)

**Comment by jdeltoft** [comment]
> This comment as a command (please make sure you understand what you are deleting):
> 
> ```
> flutter clean && rm ios/Podfile ios/Podfile.lock pubspec.lock && rm -rf ios/Pods ios/Runner.xcworkspace && flutter run
> ```

Yea, I agree.  don't run random rm -rf commands you don't fully understand.  I moved these things into a temp "cleanup" folder instead of removing them.  But yes, this did fix my issue as well.  Thank you all for your thread here!!

**Comment by sgehrman** [comment]
doesn't work for me

**Comment by Emadeddin-eibo** [comment]
None of the above has worked for me.

A simple solution is to create a new project, copy the lib & pubspec files, and run it. It worked for me.

**Comment by MytQwayNode** [comment]
Same issue for me using Xcode 11.4 and cloud_firestore: ^0.13.0+1

**Comment by ShizaS** [comment]
Im having the similar issue and its been days that i cant figure out the issue.Any help will be appreciated.
Undefined symbols for architecture armv7:
  "_FlutterMethodNotImplemented", referenced from:
      -[FLTConnectivityPlugin handleMethodCall:result:] in FLTConnectivityPlugin.o
  "_OBJC_CLASS_$_FlutterEventChannel", referenced from:
      objc-class-ref in FLTConnectivityPlugin.o
  "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
      objc-class-ref in FLTConnectivityPlugin.o
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by Greatcallie** [comment]
> 删除Podfile，Podfile.lock，Pods文件夹，Runner.xcworkspace，让flutter重新生成这几个文件，就好了

Sorry but I don't understand this, can someone explain?

**Comment by SeanZom** [comment]
> > 删除Podfile，Podfile.lock，Pods文件夹，Runner.xcworkspace，让flutter重新生成这几个文件，就好了
> 
> Sorry but I don't understand this, can someone explain?

same as https://github.com/flutter/flutter/issues/41900#issuecomment-601599410

**Comment by BayShubar** [comment]
I solved my problem by **File-> Workspace Settings**  changing build system to New Build System

**Comment by peterWilson34** [comment]
fixed for me by updating flutter to Flutter (Channel stable, 1.20.2)

**Comment by tudor07** [comment]
Still having this problem

**Comment by fullflash** [comment]
same issue.
done all cleaning upgrade etc still unable to build with latest mac and xcode 12.5 and flutter 2.2
```
 ld: warning: Could not find or use auto-linked framework 'Flutter'
               Undefined symbols for architecture x86_64:
                 "_OBJC_CLASS_$_FlutterError", referenced from:
                     objc-class-ref in firebase_core(FLTFirebasePlugin.o)
                     objc-class-ref in firebase_dynamic_links(FLTFirebaseDynamicLinksPlugin.o)
                     objc-class-ref in google_sign_in(FLTGoogleSignInPlugin.o)
                     objc-class-ref in permission_handler(PermissionHandlerPlugin.o)
                 "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
                     objc-class-ref in firebase_analytics(FLTFirebaseAnalyticsPlugin.o)
                     objc-class-ref in firebase_auth(FLTFirebaseAuthPlugin.o)
                     objc-class-ref in firebase_core(FLTFirebaseCorePlugin.o)
                     objc-class-ref in firebase_dynamic_links(FLTFirebaseDynamicLinksPlugin.o)
                     objc-class-ref in firebase_messaging(FLTFirebaseMessagingPlugin.o)
                     objc-class-ref in google_sign_in(FLTGoogleSignInPlugin.o)
                     objc-class-ref in permission_handler(PermissionHandlerPlugin.o)
                     ...
                 "_OBJC_CLASS_$_FlutterEventChannel", referenced from:
                     objc-class-ref in firebase_auth(FLTFirebaseAuthPlugin.o)
                 "_FlutterMethodNotImplemented", referenced from:
                     -[FLTFirebaseAnalyticsPlugin handleMethodCall:result:] in firebase_analytics(FLTFirebaseAnalyticsPlugin.o)
                     -[FLTFirebaseAuthPlugin handleMethodCall:result:] in firebase_auth(FLTFirebaseAuthPlugin.o)
                     -[FLTFirebaseCorePlugin handleMethodCall:result:] in firebase_core(FLTFirebaseCorePlugin.o)
                     -[FLTFirebaseDynamicLinksPlugin handleMethodCall:result:] in firebase_dynamic_links(FLTFirebaseDynamicLinksPlugin.o)
                     -[FLTFirebaseMessagingPlugin handleMethodCall:result:] in firebase_messaging(FLTFirebaseMessagingPlugin.o)
                     -[FLTGoogleSignInPlugin handleMethodCall:result:] in google_sign_in(FLTGoogleSignInPlugin.o)
                     -[PermissionHandlerPlugin handleMethodCall:result:] in permission_handler(PermissionHandlerPlugin.o)
                     ...
               ld: symbol(s) not found for architecture x86_64

```

**Comment by yxwandroid** [comment]
有相同问题

**Comment by dafinoer** [comment]
i stil get error

```
Undefined symbols for architecture arm64:
      "_OBJC_CLASS_$_FlutterMethodChannel", referenced from:
          objc-class-ref in FLTSharedPreferencesPlugin.o
      "_FlutterMethodNotImplemented", referenced from:
          ___52+[FLTSharedPreferencesPlugin registerWithRegistrar:]_block_invoke in FLTSharedPreferencesPlugin.o
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    Command CompileSwift failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description


```

**Comment by tudor07** [comment]
@Charles1198 can you please re-open the issue since this is not fixed?

**Comment by kasvith** [comment]
I also get this

**Comment by dafinoer** [comment]
please re-open this issue @Charles1198

**Comment by twocolors** [comment]
I also get this, please re-open this issue @Charles1198

**Comment by Akshar-Patel** [comment]
The following steps work for me for **Apple M1**:

**1. Clean up**
```
flutter clean && \
rm -rf ios/Pods \
rm -rf ios/Flutter/Flutter.podspec
```
**2. Get pods**
```
cd ios \
arch -x86_64 pod install //For Intel Macs, just use "pod install"
``` 

**3. Run the app (_IMPORTANT_: Running from Xcode might fail)**
```
flutter run
 
// --flavor and -d args are required if multiple flavors are configured 
// and multiple devices are connected. 
```
